internal/service/relationships.go

```diff
@@ -81,14 +91,18 @@ func (s *RelationshipsService) ReadTuples(req *pb.ReadTuplesRequest, conn pb.Kes
 }
 
 func (s *RelationshipsService) DeleteTuples(ctx context.Context, req *pb.DeleteTuplesRequest) (*pb.DeleteTuplesResponse, error) {
-	s.log.Debugf("Delete tuples request: %v", req)
+	v, err := protovalidate.New()
+	if err != nil {
+		s.log.Errorf("failed to initialize validator: ", err)
+		return nil, errors.BadRequest("Invalid request", err.Error())
+	}
 
-	if err := req.ValidateAll(); err != nil {
+	if err = v.Validate(req); err != nil {
```

Comment:
You know, we probably don't need to explicitly call Validate at all anymore. Calling it from the validation middleware _should_ be enough, no?

Reply:
I thought about that and for grpc it didn't require changing much, but I couldn't get the validator to work for http requests so I decided to leave it there. I'll take another look because it would make future changes easier

Reply:
Oh, I think I see what's happening. So, it's probably not using the validation middleware at all atm - the middleware is looking for the message to be convertible to an interface with a Validate() method on it. protoc-gen-validate used to (try to) generate that, but protovalidate does not. 

We kind of have a choice to make here- 

We could take the validation middleware out and handle it on a per-operation basis in the service layer, which is essentially what's happening now. 

We could make an alternative Kratos middleware (they're very small, see internal/server/middleware/validation.go) that calls v.Validate(m) instead of m.Validate()

We could add (or, preferably generate) a file in the api/kessel/relations/v1beta1 folder that adds Validate methods to all the messages that, in turn, just calls v.Validate(m) so they still meet the contract Kratos is expecting

Reply:
Option #2 of making an alternative Kratos middleware seemed to be the best of both worlds, where instead of having to constantly add validation each service, there's a single layer that validates for all requests. There was an initial issue in setting this up with some test cases that were previously unintentionally testing the validation portion. These tests were removed and test cases for the validation middleware itself were added to supplement the loss.